Merge latest changes

andreas83 opened the pull request.

Commit:
added newline rendering #18

Commit:
replace all, not only the first occurrence of a newline

Commit:
added time information about each post

andreas83 committed:
Added full path to og meta tag img

andreas83 committed:
Optimized OG Tag parser, fixed some undefined notices

andreas83 committed:
Allow posting stuff with api key authentication

andreas83 committed:
Allow posting with api key authentication

andreas83 committed:
only use base path if img src is not absoloute

andreas83 committed:
added basic notifications support via websocket (notification on score, comment) 
todo notify user while something is new

Commit:
changed the layout of notification table 
added from / to user_id

andreas83 committed:
fixed from / to user_id

chilimatic commented on app/controller/DataController.php:
```diff
@@ -96,7 +96,10 @@ function stream($request=false) {
 
-        if (isset($_POST) && !empty($_POST) && Helper::isUser() && !isset($_POST['wayback'])) {
+        if (
+            isset($_POST) && !empty($_POST) &&  
```
das isset && !empty ist redundant !empty erfüllt eigentlich beides :)

andreas83 replied:
Die Methode sollte generell über arbeitet werden. Aber isset und empty sind unterschiedliche Dinge, eine variabel kann gesetzt und trotzdem empty sein.

chilimatic replied:
schon klar du brauchst aber wegen dem logischen && das isset() nicht :)

chilimatic commented on app/controller/DataController.php:
```diff
@@ -210,6 +220,20 @@ function comment($request) {
             $comment->comment = $_POST['text'];
             $comment->user_id = $_SESSION['login'];
             $comment->save();
+            
```
magst das ned in eine "service klasse" (stateless) auslagern ? dann ersparst die dir die redundanz und musst den code nur auf 1 stelle warten :D

andreas83 replied:
Würde gerne als nächstes ein intensives refactoring machen. Deine Ideen würde ich dabei gerne berücksichtigen. Als erstes wird es mal ein eigenen brunch geben und wir starten mal mit @todo tags um die Problemfelder zu identifizieren.

chilimatic replied:
klingt gut :) so lang das was ich sag halt sinn macht :)

chilimatic commented on app/controller/DataController.php:
```diff
@@ -406,13 +444,24 @@ function metadata() {
              $data['og_title'] =  $dom->getElementsByTagName('title')->item(0)->textContent;
         }
-        if(!isset($data['og_description'])){
+        if(!isset($data['og_description']) && isset($data['alt_description']) && !empty($data['alt_description'])){
```
und noch einmal !empty beinnhaltet isset :)

chilimatic commented on app/controller/DataController.php:
```diff
@@ -165,9 +175,9 @@ function stream($request=false) {
             $this->addHeader('<meta property="og:type" content="website" />');
             if(isset($media->img[0]))
-                $this->addHeader('<meta property="og:image" content="'.Config::get("upload_address").$media->img[0].'"/>');
+                $this->addHeader('<meta property="og:image" content="'.Config::get("address").Config::get("upload_address").$media->img[0].'"/>');
```
$content = Config::get("address").Config::get("upload_address"); dann könntest du dir die funktionsaufrufe sparen. das ist zwar mikro optimierung wäre aber in dem fall auch einfacher zu warten da du in der nächsten condition nicht mit elseif arbeitest sondern mit einem if.

chilimatic commented on app/controller/DataController.php:
```diff
@@ -406,13 +444,24 @@ function metadata() {
+            
+            $imgSrc=$dom->getElementsByTagName('img')->item(0)->attributes->getNamedItem("src")->value;
+            
+            if(substr($imgSrc, 0, 4)=="http")
```
was ist mit https ?

andreas83 replied:
Ist in dem Fall auch gecovert, hier ging es um das unterscheiden ob relativ oder absolute Pfade. Allerdings werden https Bilder vmtl auf dmdn nicht angezeigt, aber aus anderen gründen

chilimatic replied:
ahh cool thx :) :+1:

chilimatic commented on app/lib/Helper.php:
```diff
@@ -45,8 +45,19 @@ public static function cssScript($params)
+        {
+            $user=new User;
+            $res=$user->getUserbyAPIKey($_POST['api_key']);
+            if(count($res)>0)
```
könntest eigentlich mit return (bool) $res; vereinfachen :) bzw mim ternary (bla) ? bla : blub (geschmack)

chilimatic commented on app/server/runnotificationServer.php:
```diff
@@ -0,0 +1,52 @@
+      {
+          $notifications = new Notification;
+          $res=$notifications->getNotifications($data->auth_cookie);
+          var_dump($data);
```
<3

andreas83 replied:
;)

chilimatic commented on app/server/websocketServer.php:
```diff
@@ -0,0 +1,607 @@
+
+  function __construct($addr, $port, $bufferLength = 2048) {
+    $this->maxBufferSize = $bufferLength;
+    $this->master = socket_create(AF_INET, SOCK_STREAM, SOL_TCP)  or die("Failed: socket_create()");
```
ich weiß die ist oldschool aber wie wär's mit exceptions ? :D sind irgendwie besser zum tracken :) vor allem kannst diese evtl mit monolog oder einem anderen log stream einfach in kibana oder ein tool deiner wahl streamen :)

chilimatic commented on app/server/websocketServer.php:
```diff
@@ -0,0 +1,607 @@
+      $read = $this->sockets;
+      $write = $except = null;
+      $this->_tick();
+      $this->tick();
```
ist es wichtig das man eine funktion die nichts macht regelmässig aufruft ? :)

andreas83 replied:
Ich dachte das ist die Idee von einem daemon

chilimatic replied:
$this->tick() ist laut source code empty deswegen frag ich :)

chilimatic commented on app/server/websocketServer.php:
```diff
@@ -0,0 +1,607 @@
+<?php
```
wieso hast ihn ned einfach geforked oder als submodule verwendet ? bzw ich hätt zeroMQ verwendet das skaliert und ist distributed und gibt's als php extension :)

andreas83 committed:
Moved Socket Server class to lib folder 
Mobile version -> changed header to only one collumn
Removed registration wall
added default/fallback picture
changed notification output slightly

andreas83 committed:
readede anonymous posts

Commit:
added simple spam bot prevention system

andreas83 committed:
webm now stored local 
fixed og tag path

Commit:
added custom css for profile page
fixed responsive video
fixed upload address

Commit:
custom css only on profile and settings

andreas83 committed:
added show some random images on login/register and side menu

Commit:
added some more rand images

Commit:
added filter for nsfw content

Commit:
removed wayback function in favor of new random function
